libslide/src/scanner/types.rs

```diff
@@ -0,0 +1,144 @@
+// A definition of a token - used in the scanner
+// Written by Luke Bhan, 2/19/2020
+//
+//
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum TokenType{
+    // Stores a floating point number as dp
+    Num,
+
+    // Stores an int - signed
+    Int,
+
+    // Identifies addition - stored as a char
+    Plus,
+
+    // Identifies subtraction - stored as a char
+    Minus, 
+
+    // Identifies multiplication - stored as a char
+    Mult,
+
+    // Identifies division - stored as a char
+    Div,
+
+    // Identifies modulo - stored as a char
+    Mod,
+
+    // Identifies exponent - stored as a char
+    Exp,
+
+    // Identifies an equal sign - stored as a char
+    Equal, 
+
+    // open parentheses (
+    OpenParen,
+
+    // close parentheses )
+    CloseParen, 
+
+    // open bracket [
+    OpenBra, 
+    
+    // close bracket ]
+    CloseBra,
+
+    // empty token
+    Empty
+}
+
+// This will hold are token data
+#[derive(Copy, Clone, Debug)]
+pub struct Token {
+    pub token: TokenType,
+    pub integer: i64,
+    pub float: f64,
```

Comment:
I think it might be better just to store the token value as a string. We can unwrap the actual token value later on. Otherwise if we add variables, then we will also have to include `string` as data member here.

Reply:
Another option is to create multiple token types with one base algebraic type. What do you think?

Reply:
what does it mean to have a base algebraic type

Reply:
i was just gonna do it how you did it in rcalc

Reply:
That’s what it means